test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs

```diff
@@ -521,6 +521,113 @@ public void MultilineTextTemplate_ReturnsTextArea()
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        [Fact]
+        public void PasswordTemplate_ReturnsInputElement_IgnoresFormattedModelValue()
+        {
+            // Arrange
+            var expected = "<input class=\"HtmlEncode[[text-box single-line password]]\" " +
+                "id=\"HtmlEncode[[FieldPrefix]]\" name=\"HtmlEncode[[FieldPrefix]]\" " +
+                "type=\"HtmlEncode[[password]]\" />";
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper<string>(model: null);
+            var viewData = helper.ViewData;
+            var templateInfo = viewData.TemplateInfo;
+            templateInfo.HtmlFieldPrefix = "FieldPrefix";
+
+            templateInfo.FormattedModelValue = "Formatted string";
+
+            // Act
+            var result = DefaultEditorTemplates.PasswordTemplate(helper);
+
+            // Assert
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        [Fact]
+        public void PasswordTemplate_ReturnsInputElement_IgnoresModelState()
+        {
+            // Arrange
+            var expected = "<input class=\"HtmlEncode[[text-box single-line password]]\" " +
+                "id=\"HtmlEncode[[FieldPrefix]]\" name=\"HtmlEncode[[FieldPrefix]]\" " +
+                "type=\"HtmlEncode[[password]]\" />";
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper<string>(model: null);
+            var viewData = helper.ViewData;
+            var templateInfo = viewData.TemplateInfo;
+            templateInfo.HtmlFieldPrefix = "FieldPrefix";
+
+            var modelState = viewData.ModelState;
+            modelState.SetModelValue("FieldPRefix", "Raw model string", "Attempted model string");
+
+            // Act
+            var result = DefaultEditorTemplates.PasswordTemplate(helper);
+
+            // Assert
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        [Fact]
+        public void PasswordTemplate_ReturnsInputElement_IgnoresViewData()
+        {
+            // Arrange
+            var expected = "<input class=\"HtmlEncode[[text-box single-line password]]\" " +
+                "id=\"HtmlEncode[[FieldPrefix]]\" name=\"HtmlEncode[[FieldPrefix]]\" " +
+                "type=\"HtmlEncode[[password]]\" />";
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper<string>(model: null);
+            var viewData = helper.ViewData;
+            var templateInfo = viewData.TemplateInfo;
+            templateInfo.HtmlFieldPrefix = "FieldPrefix";
+
+            viewData["FieldPrefix"] = "ViewData string";
+
+            // Act
+            var result = DefaultEditorTemplates.PasswordTemplate(helper);
+
+            // Assert
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        [Fact]
+        public void PasswordTemplate_ReturnsInputElement_UsesHtmlAttributes()
```

Comment:
This is really the only test you need. The others aren't directly related to this issue are they?

Reply:
FYI, I consolidated the other tests into `PasswordTemplate_ReturnsInputElement_IgnoresValues()`, setting everything in that. Mixing this one into that test case could be fooled by precedence between the value sources.